packages/pigeon/lib/swift_generator.dart

```diff
@@ -607,6 +580,63 @@ import FlutterMacOS
     indent.newln();
   }
 
+  void _writeDecodeCasting(
+    Root root,
```

Comment:
This method is just a relocation and simplification of code from other parts of the generator that writes decoding and casting. That code was mixed in location and required a bit of duplicated code, so I moved it into a method for consistency.